pkg/distributor/distributor.go

```diff
@@ -684,30 +706,31 @@ func (d *Distributor) sendStreamsErr(ctx context.Context, ingester ring.Instance
 }
 
 type labelData struct {
+	ls     labels.Labels
 	labels string
 	hash   uint64
 }
```

Comment:
could we modify this struct to just have the `ls`? or do we call access the `labels` enough that `ls.String()` would be too expensive?